wemake_python_styleguide/visitors/ast/conditions.py

```diff
@@ -57,6 +57,49 @@ def _get_duplicate_names(variables: List[Set[str]]):
     )
 
 
+@final
+class IfExpressionVisitor(BaseNodeVisitor):
```

Comment:
This code duplicates the existing one. You have to reuse the existing code.

Reply:
I was running into problems with flake8 requiring me to have less than 8 member functions for a class, so I was forced to create a new one.

Reply:
Creating new class does not mean that you have to duplicate the code.
There are other options:
1. Extracting new functions
2. Composition
3. Inheritance
4. Possibly others

Choose the one which fits best.